content/algorithms/dag/index.md

```diff
@@ -0,0 +1,216 @@
+---
+jupytext:
+  notebook_metadata_filter: all
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.11.1
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+language_info:
+  codemirror_mode:
+    name: ipython
+    version: 3
+  file_extension: .py
+  mimetype: text/x-python
+  name: python
+  nbconvert_exporter: python
+  pygments_lexer: ipython3
+  version: 3.9.2
+---
+
+# Tutorial: Directed Acyclic Graphs
+
+In this tutorial, we will explore the algorithms related to a directed acyclic graph
+(or a "dag" as it is sometimes called) implemented in networkx under `networkx/algorithms/dag.py`.
+
+First of all, we need to understand what a directed graph is.
+
+## Directed Graph
+
+### Example
+
+```{code-cell} ipython3
+# import some useful libraries
+%matplotlib inline
+import networkx as nx
+import matplotlib.pyplot as plt
+```
+
+```{code-cell} ipython3
+# create DiGraph
+triangle_graph = nx.from_edgelist([(1, 2), (2, 3), (3, 1)], create_using=nx.DiGraph)
+```
+
+```{code-cell} ipython3
+# draw `triangle_graph`
+nx.draw_planar(triangle_graph,
+    with_labels=True,
+    node_size=1000,
+    node_color="#ffff8f",
+    width=0.8,
+    font_size=14,
+)
+```
+
+### Definition
+
+In mathematics, and more specifically in graph theory,
+a directed graph (or DiGraph) is a graph that is made up of a set of vertices
+connected by directed edges often called arcs.
+
+## Directed Acyclic Graph
+
+### Example
+
+```{code-cell} ipython3
+# load Professor Bumstead clothing graph
+clothing_graph = nx.read_graphml(f"data/clothing_graph.graphml")
+```
+
+```{code-cell} ipython3
+# set figure size
+plt.figure(figsize=(12, 12), dpi=150)
+
+# draw `clothing_graph`
+nx.draw_planar(clothing_graph,
+    arrowsize=12,
+    with_labels=True,
+    node_size=8000,
+    node_color="#ffff8f",
+    linewidths=2.0,
+    width=1.5,
+    font_size=14,
+)
+```
+
+Here is an example that arises when Professor Bumstead gets dressed in the morning.
+The professor must do certain garments before others (e.g., socks before shoes).
+Other items may be put on in any order (e.g., socks and pants).
+
+A directed edge $(u, v)$ in the example indicates that garment $u$
+must be donned before garment $v$.
+
+For example, the `clothing_graph` is a DAG.
+
+```{code-cell} ipython3
+# check if the `clothing_graph` is DAG
+print(nx.is_directed_acyclic_graph(clothing_graph))
+```
```

Comment:
````suggestion
```{code-cell} ipython3
nx.is_directed_acyclic_graph(clothing_graph)
```
````

A few suggestions that are more opinions than anything else, so feel free to ignore:
 1. I think the comments in these sections are not really necessary, as they are basically restatements of the text above the code cell. IMO the comments actually impede the flow of the notebook as the reader ends up reading essentially the same thing twice in a row.
 2. The `print` function is unnecessary for the output to render since these are notebooks, so I have a slight preference for removing the explicit call to print. Removing nested function calls helps keep the focus on the `nx` function itself.

As mentioned - feel free to ignore these as they are only very minor suggestions. If you do decide to accept them however, note that these suggestions also apply to the code blocks below!